maint/gen_abi.py

```diff
@@ -43,12 +43,12 @@ def gen_mpi_abi_internal_h(out):
             elif RE.match(r'\s*int MPI_(SOURCE|TAG|ERROR);', line):
                 # no need to rename status fields
                 out.append(line.rstrip())
-            elif RE.match(r'\s*int (reserved|internal)\[(\d+)\];', line):
+            elif RE.match(r'\s*int MPI_(reserved|internal)\[(\d+)\];', line):
                 n = int(RE.m.group(2))
                 out.append("    int count_lo;")
                 out.append("    int count_hi_and_cancelled;")
                 out.append("    int reserved[%d];" % (n - 2))
-            elif RE.match(r'#define\s+(MPI_\w+)\s+\((MPI_\w+)\)\s*(0x\S+)', line):
+            elif RE.match(r'#define\s+(MPI_\w+)\s+\(\((MPI_\w+)\)\s*(0x\S+)\)', line):
```

Comment:
How about `\(?\((MPI_\w+)\)\s*(0x\S+)\)?`? I hate to see the code to be too dependent on ad hoc upstreams :)

Reply:
Definitely good point.

Reply:
But if we get serious about it, then we need a bunch of `\s*` before/after all `\(` and `\)`.

Reply:
Well, let's draw a line in the sand between getting serious about it and good for now :)